app/calendar/calendar-heatmap.tsx

```diff
@@ -46,23 +46,32 @@ export default function HeatMap({ startDate, endDate, events }) {
     for (var event of events) {
       event.date_time = new Date(event.date_time);
     }
+
     let newValues = [];
+    let map = new Map<string, number>();
     for (
-      var d = new Date(startDate);
-      d <= endDate;
-      d.setDate(d.getDate() + 1)
+      var D = new Date(startDate);
```

Comment:
Let's maybe use `d` instead of `D`, or even better maybe let's use `date`? a variable name should always start with lowercase. Additionally check the recommendations from mozilla [here](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Writing_style_guide/Code_style_guide/JavaScript#variable_names)